diff --git a/boards/catie/zest_core_stm32h753zi/zest_core_stm32h753zi.dts b/boards/catie/zest_core_stm32h753zi/zest_core_stm32h753zi.dts
index dd29fcc..9668452 100644
--- a/boards/catie/zest_core_stm32h753zi/zest_core_stm32h753zi.dts
+++ b/boards/catie/zest_core_stm32h753zi/zest_core_stm32h753zi.dts
@@ -148,15 +148,25 @@
 };

 &mac {
-	status = "okay";
-	pinctrl-0 = <&eth_mdc_pc1
-		     &eth_rxd0_pc4
-		     &eth_rxd1_pc5
-		     &eth_ref_clk_pa1
-		     &eth_mdio_pa2
-		     &eth_crs_dv_pa7
-		     &eth_tx_en_pb11
-		     &eth_txd0_pb12
-		     &eth_txd1_pb13>;
-	pinctrl-names = "default";
-};
+    status = "okay";
+    pinctrl-0 = <&eth_rxd0_pc4
+             &eth_rxd1_pc5
+             &eth_ref_clk_pa1
+             &eth_crs_dv_pa7
+             &eth_tx_en_pb11
+             &eth_txd0_pb12
+             &eth_txd1_pb13>;
+    pinctrl-names = "default";
+};
+
+&mdio {
+    pinctrl-0 = <&eth_mdio_pa2 &eth_mdc_pc1>;
+    pinctrl-names = "default";
+
+    ethernet-phy@0 {
+        compatible = "ethernet-phy";
+        reg = <0x00>;
+        status = "okay";
+    };
+    status = "okay";
+};
\ No newline at end of file
